code/datums/components/plumbing/reaction_chamber.dm

```diff
@@ -18,10 +18,18 @@
 	if(RC.emptying)
 		return
 
-	process_request(amount = min(MACHINE_REAGENT_TRANSFER, RC.target_volume - reagents.total_volume), dir = dir)
-
-	if(RC.target_volume > round(reagents.total_volume, CHEMICAL_VOLUME_ROUNDING)) //not enough yet
-		return
+	for(var/RT in RC.required_reagents)
```

Comment:
https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names